lib/core.mli

```diff
@@ -78,6 +78,9 @@ val group_count : re -> int
 (** Return the number of capture groups (including the one
     corresponding to the entire regexp). *)
 
+val group_names : re -> (int * string) list
```

Comment:
`(string * int) list` might make more sense, to be able to use the result as an associative list.

Reply:
Indeed. Done.